python/src/scipp/plotting/model.py

```diff
@@ -104,19 +107,34 @@ def _axis_coord_and_formatter(self, dim, data_array, dim_to_shape,
 
         contains_strings = False
         contains_vectors = False
+        contains_datetime = False
+        offset = 0.0
+        coord_label = None
+        form = None
 
         has_no_coord = dim not in data_array.meta
         if not has_no_coord:
             if data_array.meta[dim].dtype == sc.dtype.vector_3_float64:
                 contains_vectors = True
             elif data_array.meta[dim].dtype == sc.dtype.string:
                 contains_strings = True
+            elif data_array.meta[dim].dtype == sc.dtype.datetime64:
+                contains_datetime = True
 
         # Get the coordinate from the DataArray or generate a fake one
         if has_no_coord or contains_vectors or contains_strings:
             coord = make_fake_coord(dim, dim_to_shape[dim] + 1)
             if not has_no_coord:
                 coord.unit = data_array.meta[dim].unit
+        elif contains_datetime:
+            nanoseconds = data_array.meta[dim].values.astype(
+                'datetime64[ns]').astype(np.int64)
```

Comment:
Why do we need to work in nanoseconds? Would potentially risk running out of range, I suppose?

Reply:
true, if ESS will still be around in 200 years...
I'm not entirely sure if matplotlib internally uses float32 or 64 internally, and whether it even matters.
Do you mean we could use float and work in e.g. microseconds?

Reply:
No, I mean why do we need to convert the unit at all?

Reply:
You mean we could just convert to integers from numpy, whatever that gives you (seconds, or days, and work with that?)

![Screenshot at 2021-03-17 09-15-47](https://user-images.githubusercontent.com/39047984/111435752-71dda400-8701-11eb-894c-ec7c898b48e8.png)

i guess because it's easier when applying the formatting if I know in advance that I have nanoseconds. But we could get the unit from numpy at the time of conversion and adjust the formatting using that?

Reply:
Yes, it should just be a simple factor.